3rdparty/openvx/include/ivx.hpp

```diff
@@ -0,0 +1,1606 @@
+    }
+
+    template<typename T0, typename T1, typename T2>
+    static Node create(vx_graph g, vx_enum kernelID,
+                       const T0& arg0, const T1& arg1, const T2& arg2)
+    {
+        std::vector<vx_reference> params;
+        params.push_back(castToReference(arg0));
+        params.push_back(castToReference(arg1));
+        params.push_back(castToReference(arg2));
+        return create(g, Kernel::getByEnum(Context::getFrom(g), kernelID), params);
+    }
+
+    template<typename T0, typename T1, typename T2, typename T3>
+    static Node create(vx_graph g, vx_enum kernelID,
+                       const T0& arg0, const T1& arg1, const T2& arg2,
+                       const T3& arg3)
+    {
+        std::vector<vx_reference> params;
+        params.push_back(castToReference(arg0));
+        params.push_back(castToReference(arg1));
+        params.push_back(castToReference(arg2));
+        params.push_back(castToReference(arg3));
+        return create(g, Kernel::getByEnum(Context::getFrom(g), kernelID), params);
+    }
+
+    template<typename T0, typename T1, typename T2, typename T3, typename T4>
+    static Node create(vx_graph g, vx_enum kernelID,
+                       const T0& arg0, const T1& arg1, const T2& arg2,
+                       const T3& arg3, const T4& arg4)
+    {
+        std::vector<vx_reference> params;
+        params.push_back(castToReference(arg0));
+        params.push_back(castToReference(arg1));
+        params.push_back(castToReference(arg2));
+        params.push_back(castToReference(arg3));
+        params.push_back(castToReference(arg4));
+        return create(g, Kernel::getByEnum(Context::getFrom(g), kernelID), params);
+    }
+
+    template<typename T0, typename T1, typename T2, typename T3, typename T4, typename T5>
+    static Node create(vx_graph g, vx_enum kernelID,
+                       const T0& arg0, const T1& arg1, const T2& arg2,
+                       const T3& arg3, const T4& arg4, const T5& arg5)
+    {
+        std::vector<vx_reference> params;
+        params.push_back(castToReference(arg0));
+        params.push_back(castToReference(arg1));
+        params.push_back(castToReference(arg2));
+        params.push_back(castToReference(arg3));
+        params.push_back(castToReference(arg4));
+        params.push_back(castToReference(arg5));
+        return create(g, Kernel::getByEnum(Context::getFrom(g), kernelID), params);
+    }
+
+    void setParameterByIndex(vx_uint32 index, vx_reference value)
+    { IVX_CHECK_STATUS(vxSetParameterByIndex(ref, index, value)); }
+};
+
+#else // not IVX_USE_CXX98
+
+class Node : public RefWrapper<vx_node>
+{
+public:
+    IVX_REF_STD_CTORS_AND_ASSIGNMENT(Node);
+
+    static Node create(vx_graph g, vx_kernel k)
+    { return Node(vxCreateGenericNode(g, k)); }
+
+    static Node create(vx_graph graph, vx_kernel kernel, std::initializer_list<vx_reference> params)
```

Comment:
Is c++98 overload with `const std::vector<vx_reference>& params` processed here?